core/templates/dev/head/components/solution_editor_directive.html

```diff
@@ -0,0 +1,103 @@
+<div style="height: 100%;">
+  <div ng-if="!solutionEditorIsOpen"
+       style="height: 100%;"
+       ng-attr-title="<[isEditable ? 'Edit solution' : '']>">
+    <div class="oppia-readonly-rule-tile"
+         ng-class="{'oppia-editable-section': isEditable}">
+      <div class="oppia-rule-preview oppia-transition-200">
+        <div class="oppia-click-to-start-editing"
+             ng-if="isEditable" ng-click="openSolutionEditor()">
+          <i ng-if="isEditable"
+             class="material-icons oppia-editor-edit-icon pull-right"
+             title="Edit Solution">&#xE254;
+          </i>
+        </div>
+        <strong>
+          <[solution.answerIsExclusive ? 'Only' : 'One']> Solution is...
+        </strong>
+        <span ng-if="objectType !== 'CodeString' &&
+                     objectType !== 'Graph' &&
+                     objectType !== 'UnicodeString'">
+          <[solution.correctAnswer]>
+        </span>
+        <span ng-if="objectType === 'CodeString'">
+          <[solution.correctAnswer.code]>
+        </span>
+        <span ng-if="objectType === 'Graph'">
+          [Graph Object]
+        </span>
+        <span ng-if="objectType === 'UnicodeString'">
+          <[solution.correctAnswer.latex]>
+        </span>
+        <br>
+        <strong>Explanation is...</strong>
+        <span angular-html-bind="solution.explanation"></span>
+      </div>
+    </div>
+  </div>
+
+  <div ng-if="isEditable && solutionEditorIsOpen">
+    <form role="form"
+          class="form-inline"
+          name="editSolutionForm"
+          ng-submit="saveThisSolution()">
+      <div class="oppia-rule-details-header">
+        <select ng-model="solution.answerIsExclusive"
+                ng-options="(item ? 'Only' : 'One') for item in [true, false]">
+        </select>
+        <strong>Solution is...</strong>
+        <div ng-if="!interactionHtml">
+          <object-editor ng-if="objectType === 'UnicodeString'"
+                         obj-type="<[objectType]>"
+                         init-args="{choices: ruleDescriptionChoices}"
+                         is-editable="true"
+                         always-editable="true"
+                         value="solution.correctAnswer.latex"
```

Comment:
Same thing -- maybe define a function on the Solution object for this so that there's not so much repetition in the HTML and so that it's testable.